pulp_file/tests/functional/api/from_pulpcore/test_crud_repos.py

```diff
@@ -225,234 +145,185 @@ def setUp(self):
-        }
-        remote = self.api_client.post(FILE_REMOTE_PATH, data=remote_attrs)
-        self.addCleanup(self.remotes_api.delete, remote["pulp_href"])
-        self.assertEqual(remote["pulp_labels"], self.pulp_labels)
+    assert json.loads(remote)["pulp_labels"] == pulp_labels
 
 
 @pytest.mark.parallel
-class RemoteFileURLsValidationTestCase(unittest.TestCase):
+def test_file_remote_url_validation(file_remote_api_client, gen_object_with_cleanup):
     """A test case that verifies the validation of remotes' URLs."""
 
-    @classmethod
-    def setUpClass(cls):
-        """Initialize class-wide variables"""
-        cls.cfg = config.get_config()
-
-        cls.api_client = api.Client(cls.cfg, api.json_handler)
-        cls.file_client = FileApiClient(cls.cfg.get_bindings_config())
-        cls.remotes_api = RemotesFileApi(cls.file_client)
-
-    def test_invalid_absolute_pathname(self):
-        """Test the validation of an invalid absolute pathname."""
-        remote_attrs = {
-            "name": utils.uuid4(),
-            "url": "file://error/path/name",
-        }
-        self.raise_for_invalid_request(remote_attrs)
-
-    def test_invalid_import_path(self):
-        """Test the validation of an invalid import pathname."""
-        remote_attrs = {
-            "name": utils.uuid4(),
-            "url": "file:///error/path/name",
-        }
-        self.raise_for_invalid_request(remote_attrs)
-
-    def raise_for_invalid_request(self, remote_attrs):
+    def raise_for_invalid_request(remote_attrs):
         """Check if Pulp returns HTTP 400 after issuing an invalid request."""
-        with self.assertRaises(ApiException) as ae:
-            remote = self.remotes_api.create(remote_attrs)
-            self.addCleanup(self.remotes_api.delete, remote.pulp_href)
-
-        self.assertEqual(ae.exception.status, 400)
-
-    def test_valid_import_path(self):
-        """Test the creation of a remote after passing a valid URL."""
-        remote_attrs = {
-            "name": utils.uuid4(),
-            "url": "file:///tmp/good",
-        }
-
-        remote = self.remotes_api.create(remote_attrs)
-        self.addCleanup(self.remotes_api.delete, remote.pulp_href)
-
-    def test_no_username_password(self):
-        """Test that the remote url can't contain username/password."""
-        remote_attrs = {
-            "name": utils.uuid4(),
-            "url": "http://elladan@rivendell.org",
-        }
-        self.raise_for_invalid_request(remote_attrs)
-
-        remote_attrs = {
-            "name": utils.uuid4(),
-            "url": "http://elladan:pass@rivendell.org",
-        }
-        self.raise_for_invalid_request(remote_attrs)
+        with pytest.raises(ApiException) as ae:
+            file_remote_api_client.create(remote_attrs)
+            assert ae.value.status == 400
+
+    # Test the validation of an invalid absolute pathname.
+    remote_attrs = {
+        "name": utils.uuid4(),
+        "url": "file://error/path/name",
```

Comment:
I know that this is just a port of the existing test, but I don't understand the difference between this and the previous assertion.

Reply:
There is one / missing. But maybe we should use
```suggestion
        "url": "file://tmp/good",
```
to only have one reason to fail.